client/src/main/scala/spatutorial/client/components/MainMenu.scala

```diff
@@ -34,27 +37,30 @@ object MainMenu {
     MenuItem(2, _ => "Flights", Icon.plane, FlightsLoc),
     MenuItem(3, buildTodoMenu, Icon.calculator, UserDeskRecommendationsLoc))
 
-  def menuItems(terminals: Seq[String]) = {
+  def menuItems(airportConfigPotMP: ModelProxy[Pot[AirportConfigHolder]]) = {
```

Comment:
what was wrong with terminals: Seq[String]. 
Pass the *smallest* *simples* surface area to the components that you can. All we need is names.